Solve DNS not resolve after network comes up.

papampi opened the pull request:
While name resolution not working check ping and restart 3main if ping is ok.

papampi committed:
Update 5watchdog

papampi committed:
Fix indent

LuKePicci commented:
Not sure how this can solve the problem. I mean, how restarting an application could restore system-wide name resolution functionality? Every time the` nc google.com 443` loop detect a connection problem it restart a new instance of the netcat program, so restarting netcat won't let netcat get network resolution fixed. How restarting the miner application could solve name resolution issues affecting netcat? I would like to see the bug bulletin @doftorul was referring to in #94 .

papampi committed:
More check on start 

Now when watchdog restarted because of DNS problem it waits 10 minutes for problem to get resolved if not will reboot.

papampi commented:
I tested it on 5 rigs, no idea whats the problem, when rigs stuck on dns resolve issue I killed miner manually after 2-3 minutes dns gets resolved.

papampi commented:
Will send you some logs via email

papampi commented:
Yup, restarted network-manager as well, no help if miner still running.

papampi commented:
We can also restart network-manager as well when restarting 3main, not tested, no idea if it solve the issue faster or not

papampi commented:
Checked your email?

LuKePicci commented:
Try adding in "NetworkManager->WiredConnection>IPv4>Additional DNS Servers" a public dns like 8.8.4.4 or 1.1.1.1 and see if this help.
What sounds really strange to me is that it become functional again after a lot of time ( 3 minutes is not a short time ).

papampi commented:
Already done that last week, all has dns "192.168.1.1, 1.1.1.1, 8.8.8.8"
Some didnt have router ip in dns list and they were showing this problem more, so added it to all but still some has this problem if network disconnects more than 10-15 minutes.

LuKePicci commented:
Ok, dnsmasq running on localhost (127.0.0.1) should be automaticaly instructed to forward to dns provided by dhcp. If your dhcp server is sending to all rigs as dns to use 192.168.1.1 and it somehow caches the failure then it may return a cannot resolve error. Try editing that field in network manager but avoiding inserting again 192.168.1.1 as an alternative since it isa already what dhcp sets as default. Conversely, if your dhcp is instructed to send something different, let's say 8.8.8.8, try setting it back to send 192.168.1.1.

papampi commented:
All rigs has static ip and dns, those without 192.168.1.1 in DNS list was showing this symptom, if I remove 192.168.1.1 all will start showing this problem again.
From 6 rigs without 192.168.1.1, only 2 still has this problem.
Have no local access to them now, will change their dns in next visit.

LuKePicci commented:
Do you feel that using only 192.168.1.1 as the only dns on all rigs would solve the problem? My rig use dhcp and its dns is also my gateway, and never experienced such a problem,
This PR has a potential pitfall: if the miner is still working fine and the internet check loop fails to connect to google with netcat for some other reasons but ping works, it is restarted without any need.

papampi commented:
I will check setting dns to only gateway in next visit.
As for potential pitfall, in first check before loop starts if netcat can't resolve then miner can't resolve too and won't start mining.
And in the main loop network check only kicks in if utilization is low and miner shows problems.
Do I miss some thing else?

papampi committed:
Update 5watchdog

Check internet and pool only if all GPU utilizations are bellow threshold

papampi committed:
Update 5watchdog

elif ping and dns both ok

papampi committed:
Update 5watchdog

Sleep 10 before reboot

LuKePicci commented:
If we assume every miner freeing up gpu resources while network connection is failing then this is fine. Does this assumption holds?

Da: Payam Nab
Inviato: mercoledì 4 luglio 2018 14:46
A: papampi/nvOC_by_fullzero_Community_Release
Cc: Luca Piccirillo; Review requested
Oggetto: Re: [papampi/nvOC_by_fullzero_Community_Release] Solve DNS notresolve after network comes up. (#116)

I will check setting dns to only gateway in next visit.
As for potential pitfall, in first check before loop starts if netcat can't resolve then miner can't resolve too and won't start mining.
And in the main loop network check only kicks in if utilization is low and miner shows problems.
Do I miss some thing else?
—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub, or mute the thread.

papampi commented:
Miner should free up gpu resources when it get no new jobs, I remember there was a bug in ethminer 14 and it was not freeing up gpus when no work was received because of no network and they fixed it.
All the miners I have seen drop GPU utilization after 2-3 minutes of no network.

LuKePicci commented on 5watchdog:
```diff
@@ -158,8 +158,28 @@ fi
+    then
+      echo "Ping and DNS are OK"
+    fi
+  fi
```
uhm… it is far too complex, simplify this by doing:
```
DNS=0
while ! nc …
do
  if (( DNS >= 20 ))
  then
    echo … waited 10 minutes ...
    reboot
    exit
  fi
  echo … failed test connection ...
  if ping …
  then
    echo … but ping is ok ...
    (( DNS++ ))
  fi
  sleep ...
done
```

papampi replied:
👍

papampi committed:
Update 5watchdog

papampi committed:
Update 5watchdog
